dissect/sql/utils.py

```diff
@@ -1,27 +1,41 @@
 import re
-from typing import Optional, Tuple
+from typing import Iterator, Optional, Tuple
 
 from dissect.sql.exceptions import InvalidSQL
 
 
-def split_sql_list(sql):
+def split_sql_list(sql: str) -> Iterator[str]:
     """Split a string on comma's (`,') while ignoring any comma's contained
     within an arbitrary level of nested braces (`( )')
     """
     level = 0
+    comment = 0
+    quote = {'"': False, "'": False, "`": False}
     line_buf = ""
+
     for char in sql:
-        if char == "(":
-            level += 1
-            line_buf += char
-        elif char == ")":
-            level -= 1
-            line_buf += char
-        elif char == "," and level == 0:
-            yield line_buf.strip()
-            line_buf = ""
-        else:
-            line_buf += char
+        if not any(quote.values()) and char == "-" and comment != 2:
```

Comment:
In the test case example, this would not parse a comment after that line, as the `'` and ``` quotes would still be set to `True` it seems?